web/docs/operators/sources/show.md

```diff
@@ -72,3 +74,9 @@ Show the version of a remote node:
 ```
 remote show version
 ```
+
+Show all aspects of a remote node:
+
+```
+show
+```
```

Comment:
This nicely illustrates the confusion.

Reply:
I think the problem more generally is that an operator should not set its location based on its arguments. Changing this to always be remote would simplify the operator a lot, but it breaks `tenzir 'show version'` without a node—which I think would be fine.

@mavam wdyt?

Reply:
>I think the problem more generally is that an operator should not set its location based on its arguments.

I lack understanding to comment on that, but...

> Changing this to always be remote would simplify the operator a lot, but it breaks tenzir 'show version' without a node—which I think would be fine.

...I find it more helpful to show the version of the `tenzir` binary than the remote node.

In general, for the CLI to be remain flexible, I would argue in the opposite direction: if possible, do *not* require a node execution context.

Reply:
The problem is that it's hard to document and to explain the expected behavior to the user. When you use `show`, the version aspect is run remotely. When you use `show version`, it is run locally. If I understood Jannis correctly, then the both of us think that this is a confusing user experience.

Reply:
Operator locations themselves are already a hard-to-grok topic, but if an operator can dynamically change its location like the `show` operator than it's hard to understand what's going on.

Reply:
I agree that this is confusing, and (now) also that location should be independent of arguments.

I still find it essential to not lose the ability to show the local version. Would anything speak against introducing `--version` to the executable? The expectation is very clear hear: just show the version of the binary.

Reply:
I find it okay-ish that the location depends on the argument. That `show version` and `show` use different locations to retrieve the version is what's confusing to me, and the same goes for every other anywhere/remote aspect. I guess most of the times the result is the same. However, `show operators` for example can now often change with `disable-plugins`.

Reply:
Are there any aspects other than `version` for which we would need a local alternative?

Reply:
If you use `tenzir` without a node, then you'd probably want `show operators` to show the local operators, for example.

Reply:
My current intuition is to just merge this as-is. I think as soon as we work on merging results of multiple operators into one (e.g., via nested pipelines) this topic will come up again anyways.

Wdyt, yes/no?

Reply:
Perhaps you just remove the "show the version of a remote node" example and we'll just glance over the fact that this is a bit inconsistent.

Reply:
Done and pushed. Resolving this for now.